pipeline.sh

```diff
@@ -33,14 +35,17 @@ stack run normalize-phi < "$I" > "$IO" \
     cat "$IO"
     exit 1
   }
+perl -i -pe 'chomp if eof' "$IO"
 
-printf "\n\nNormalizer succeeded!\n\n"
+printf "\n\nNormalizer succeeded!"
+printf "\n\n* EO expression:\n\n"
+cat app.eo
 printf "\n\n* Phi expression:\n\n"
 cat "$I"
 printf "\n\n* Normalized Phi expression:\n\n"
 cat "$IO"
 printf "\n\n* Diff:\n\n"
-diff "$I" "$IO"
+diff "$I" "$IO" || true
```

Comment:
Why `|| true`?

Reply:
`diff` exits with a non-zero code when files differ. 
Here, I compare the initial term and the normalized term, and they may differ.